citesphere/src/main/java/edu/asu/diging/citesphere/web/user/ItemController.java

```diff
@@ -35,4 +40,21 @@ public String getItem(Authentication authentication, Model model, @PathVariable(
         }
         return "auth/group/items/item";
     }
+    
+    @RequestMapping(value="/auth/group/{zoteroGroupId}/item/{itemId}")
+    public ResponseEntity<String> displayItem(Authentication authentication, Model model, @PathVariable("zoteroGroupId") String zoteroGroupId, @PathVariable("itemId") String itemId) throws GroupDoesNotExistException, CannotFindCitationException, JsonProcessingException {
+        ICitation citation = citationManager.getCitation((IUser)authentication.getPrincipal(), zoteroGroupId, itemId);
+        ObjectMapper mapper = new ObjectMapper();
+        ObjectNode linkNode = mapper.createObjectNode();
+        linkNode.put("zoteroGroupId", zoteroGroupId);
+        if (citation != null) {
+            List<String> fields = new ArrayList<>();
+            citationManager.getItemTypeFields((IUser)authentication.getPrincipal(), citation.getItemType()).forEach(f -> fields.add(f.getFilename()));
+            linkNode.put("citation", mapper.writeValueAsString(citation));
+            linkNode.put("fields", mapper.writeValueAsString(fields));
```

Comment:
let's only add valid fields from zotero to the json plus any additional fields citesphere provides (and get rid of the "fields" field). So any field that would not show up on the webpage screen, shouldn't show up here either.